### PR TITLE
fix: gen jsonSchema conditional conversion to snakecase

### LINKED ISSUE #1294: [Bug] Import JsonSchema: all props converted to snake case
`kcl import --force -m jsonschema test.json` 

I'm not sure why it converted to snake case without any condition. Maybe we could add an additional option like --snake | --camel if we want to change the case and if no option is provided, the case of properties should not be changed.

Source: https://github.com/kcl-lang/kcl-go/blob/945ea9cc4ce681c4242c575df80f1b098b2ffbd4/pkg/tools/gen/genkcl_jsonschema.go#L422

### 1. Minimal reproduce step (Required)

<details>

<summary>Json Schema</summary>

test.json
```json
{
  "$id": "https://example.com/person.schema.json",
  "$schema": "https://json-schema.org/draft/2020-12/schema",
  "title": "Person",
  "type": "object",
  "properties": {
    "firstName": {
      "type": "string",
      "description": "The person's first name."
    },
    "lastName": {
      "type": "string",
      "description": "The person's last name."
    },
    "age": {
      "description": "Age in years which must be equal to or greater than zero.",
      "type": "integer",
      "minimum": 0
    }
  }
}
```
`kcl import --force -m jsonschema test.json`
</details>

### 2. What did you expect to see? (Required)
<details>

<summary>Correct KCL Schema</summary>

```KCL
"""
This file was generated by the KCL auto-gen tool. DO NOT EDIT.
Editing this file might prove futile when you re-run the KCL auto-gen generate command.
"""

schema PersonSchema:
    r"""
    PersonSchema

    Attributes
    ----------
    firstName : str, optional
        The person's first name.
    lastName : str, optional
        The person's last name.
    age : int, optional
        Age in years which must be equal to or greater than zero.
    """

    firstName?: str
    lastName?: str
    age?: int

    check:
        age >= 0

```

</details>

### 3. What did you see instead (Required)

<details>

<summary>Snaked KCL Schema</summary>

```KCL
"""
This file was generated by the KCL auto-gen tool. DO NOT EDIT.
Editing this file might prove futile when you re-run the KCL auto-gen generate command.
"""

schema PersonSchema:
    r"""
    PersonSchema

    Attributes
    ----------
    first_name : str, optional
        The person's first name.
    last_name : str, optional
        The person's last name.
    age : int, optional
        Age in years which must be equal to or greater than zero.
    """

    first_name?: str
    last_name?: str
    age?: int

    check:
        age >= 0

```
</details>

### 4. What is your KCL components version? (Required)

0.8.7-darwin-arm64

### PR DESCRIPTION
**Pull request by shruti2522**
#### 1. Does this PR affect any open issues?(Y/N) and add issue references (e.g. "fix #123", "re #123".):

- [ ] N
- [x] Y 
 fix [#1294](https://github.com/kcl-lang/kcl/issues/1294) in kcl-lang/kcl

#### 2. What is the scope of this PR (e.g. component or file name):

kcl-lang/kcl/pkg/tools/gen

#### 3. Provide a description of the PR(e.g. more details, effects, motivations or doc link):

- [x] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Other

#### 4. Are there any breaking changes?(Y/N) and describe the breaking changes(e.g. more details, motivations or doc link):

- [ ] N
- [x] Y 
Output schema conditionally converted to snake case.

#### 5. Are there test cases for these changes?(Y/N) select and add more details, references or doc links:

- [ ] Unit test
- [ ] Integration test
- [ ] Benchmark (add benchmark stats below)
- [ ] Manual test (add detailed scripts or steps below)
- [x] Other

### CODE REVIEW

File: pkg/tools/gen/genkcl_jsonschema.go [hunk 1]
```diff
@@ -419,7 +420,11 @@ func convertSchemaFromJsonSchema(ctx *convertContext, s *jsonschema.Schema, name
 	if result.HasIndexSignature && result.IndexSignature.validation != nil {
 		result.Validations = append(result.Validations, *result.IndexSignature.validation)
 	}
-	result.property.Name = strcase.ToSnake(result.Name)
+	if ctx.convertToSnake {
```
Comment (borgius) [inline pkg/tools/gen/genkcl_jsonschema.go]: Can we have two options? One convert to snake, second convert to camel, but if no option use original name

Comment (shruti2522) [reply pkg/tools/gen/genkcl_jsonschema.go]: done.

File: pkg/tools/gen/genkcl_jsonschema.go [hunk 2]
```diff
@@ -419,7 +430,15 @@ func convertSchemaFromJsonSchema(ctx *convertContext, s *jsonschema.Schema, name
 	if result.HasIndexSignature && result.IndexSignature.validation != nil {
 		result.Validations = append(result.Validations, *result.IndexSignature.validation)
 	}
-	result.property.Name = strcase.ToSnake(result.Name)
+	switch ctx.casingOption {
```
Comment (Peefy) [inline pkg/tools/gen/genkcl_jsonschema.go]: The CI faild. Could you please fix it and add more unit tests for different options like the related issue and PR: https://github.com/kcl-lang/kcl-go/pull/296? Thank you!

File: pkg/tools/gen/genkcl_jsonschema.go [hunk 3]
```diff
@@ -15,14 +15,25 @@ import (
 	"kcl-lang.io/kcl-go/pkg/logger"
 )
 
+type casingOption int
+
+var propertyName string
```
Comment (Peefy) [inline pkg/tools/gen/genkcl_jsonschema.go]: Use a local var instead of a global one.

File: pkg/tools/gen/genkcl_jsonschema.go [hunk 4]
```diff
@@ -15,14 +15,25 @@ import (
 	"kcl-lang.io/kcl-go/pkg/logger"
 )
 
+type casingOption int
```
Comment (Peefy) [inline pkg/tools/gen/genkcl_jsonschema.go]: s/casingOption/castingOption

File: pkg/tools/gen/genkcl_test.go [hunk 5]
```diff
@@ -13,19 +13,20 @@ import (
...
-package main
-type employee struct {
-	name        string            ` + "`kcl:\"name=name,type=str\"`" + `           // kcl-type: str
-	age         int               ` + "`kcl:\"name=age,type=int\"`" + `            // kcl-type: int
-	friends     []string          ` + "`kcl:\"name=friends,type=[str]\"`" + `      // kcl-type: [str]
-	movies      map[string]*Movie ` + "`kcl:\"name=movies,type={str:Movie}\"`" + ` // kcl-type: {str:Movie}
-	bankCard    int               ` + "`kcl:\"nam=bankCard,type=int\",abc:\"name=bankCard,type=int\"`" + `       // kcl-type: int
-	nationality string            ` + "`kcl:\"name=nationality,type=str\"`" + `    // kcl-type: str
-}`
+	// 	const code := `
```
Comment (Peefy) [inline pkg/tools/gen/genkcl_test.go]: Why change this code here?

File: pkg/tools/gen/genkcl_test.go [hunk 6]
```diff
@@ -134,17 +136,21 @@ func TestGenKclFromJsonSchema(t *testing.T) {
 	for _, caseFile := range caseFiles {
 		input := filepath.Join(casesPath, caseFile.Name(), "input.json")
 		expectFilepath := filepath.Join(casesPath, caseFile.Name(), "expect.k")
+
+		castingOption := originalName
+
 		cases = append(cases, testCase{
```
Comment (Peefy) [inline pkg/tools/gen/genkcl_test.go:128]: There are some sill test cases failed. You can rebase the main branch and update all test cases.

File: pkg/tools/gen/genkcl_jsonschema.go [hunk 7]
```diff
@@ -43,13 +43,14 @@ type convertResult struct {
 }
 
 func convertPropertyName(name string, option castingOption) string {
+	name = strings.ReplaceAll(name, `"`, "")
```
Comment (Peefy) [inline pkg/tools/gen/genkcl_jsonschema.go]: Dot not replace `"`, "cancel-in-progress" it is a valid identifier.

**Comment by shruti2522** [comment]
What's the update on this @Peefy , should I keep the quotes in the original name?

**Comment by Peefy** [comment]
> What's the update on this @Peefy , should I keep the quotes in the original name?

Yes, for the attribute name `a-b`, its form should be `"a-b": <some type annotation>` in the test case instead of `a-b: <some type annotation>`.

**Comment by shruti2522** [comment]
> > What's the update on this @Peefy , should I keep the quotes in the original name?
> 
> Yes, for the attribute name `a-b`, its form should be `"a-b": <some type annotation>` in the test case instead of `a-b: <some type annotation>`.

Got it. Would push the changes soon.

**Comment by Peefy** [comment]
> > > What's the update on this @Peefy , should I keep the quotes in the original name?
> > 
> > 
> > Yes, for the attribute name `a-b`, its form should be `"a-b": <some type annotation>` in the test case instead of `a-b: <some type annotation>`.
> 
> Got it. Would push the changes soon.

OK, I've opened a sample PR, you can refer this. https://github.com/kcl-lang/kcl-go/pull/298

**Comment by shruti2522** [comment]
> OK, I've opened a sample PR, you can refer this. #298

Thank you @Peefy &#x1F90D;

**Comment by shruti2522** [comment]
Updated the test cases @Peefy

**Comment by Peefy** [comment]
> Updated the test cases @Peefy

@shruti2522 Pls rebase the main branch. Thank you!

**Comment by shruti2522** [comment]
done.

**Review by Peefy — APPROVED** [review]
LGTM

**Pull request merged**